### PR TITLE
Unapologetic Research Buffs: A Rope of Sand

### PR DESCRIPTION
**Pull request by Snakebittenn**
- rscadd: "Adds stasis bags, kinetic analyzers, gas masks, mini extinguishers, extinguisher refills, hoist kits, jetpacks, mobile ladders, heavy flashlights, tethering devices, hand teleporters, tactical masks, NVGs, gravity guns, and ammo displays to the protolathe."
  - rscadd: "Adds brass knuckles to the autolathe's hacked section."

### CODE REVIEW

**Comment by Snakebittenn** [comment]
!review

**Comment by Snakebittenn** [comment]
Update: Adds maglights and stunrods. Yanks RFDs out of the autolathe, puts them in protolathe.

**Review by Arrow768 — APPROVED** [review]
Technically alright,
but not in favor as all of these things can be ordered via cargo.
There is no need to construct all these things on the station.

Will be discussed between the maintainers.

**Review by Geevies — APPROVED** [review]

File: code/modules/research/designs/protolathe/medical_designs.dm [hunk 1]
```diff
@@ -110,6 +110,13 @@
 	materials = list(DEFAULT_WALL_MATERIAL = 3000, MATERIAL_PHORON = 3000, MATERIAL_DIAMOND = 500)
 	build_path = /obj/item/reagent_containers/personal_inhaler_cartridge/bluespace
 
+/datum/design/item/medical/stasisbag
+	name = "Stasis Bag"
+	desc = "A folded, reusable bag designed to prevent additional damage to an occupant, especially useful if short on time or in a hostile enviroment."
+	req_tech = list(TECH_BIO = 6)
+	materials = list(DEFAULT_WALL_MATERIAL = 3000, MATERIAL_GOLD = 2000, MATERIAL_DIAMOND = 500)
+	build_path = /obj/item/bodybag/cryobag
+
```
Comment (NonQueueingMatt) [inline code/modules/research/designs/protolathe/medical_designs.dm:119]: I'm against these ever being constructible via R&D. Get rid of these.

File: code/modules/research/designs/protolathe/tool_designs.dm [hunk 2]
```diff
@@ -107,4 +107,82 @@ datum/design/item/tool/advanced_light_replacer
...
+	desc = "A setup kit for a hoist that can be used to lift things. The hoist will deploy in the direction you're facing."
+	req_tech = list(TECH_ENGINEERING = 4)
+	materials = list(DEFAULT_WALL_MATERIAL = 1500)
+	build_path = /obj/item/hoist_kit
+
+/datum/design/item/tool/jetpack
+	desc = "A tank of compressed gas for use as propulsion in zero-gravity areas. Use with caution."
+	req_tech = list(TECH_ENGINEERING = 4, TECH_MAGNET = 4)
+	materials = list(DEFAULT_WALL_MATERIAL = 1500, MATERIAL_GLASS = 200)
+	build_path = /obj/item/tank/jetpack
```
Comment (NonQueueingMatt) [inline code/modules/research/designs/protolathe/tool_designs.dm:140]: Too cheap. Should require mining materials and higher tech.

File: code/modules/research/designs/protolathe/tool_designs.dm [hunk 3]
```diff
@@ -107,4 +107,82 @@ datum/design/item/tool/advanced_light_replacer
...
+	desc = "A device used by explorers to keep track of partners by way of electro-tether."
+	req_tech = list(TECH_MAGNET = 5)
+	materials = list(DEFAULT_WALL_MATERIAL = 1500, MATERIAL_SILVER = 500)
+	build_path = /obj/item/tethering_device
+
+/datum/design/item/tool/hand_tele
+	desc = "A hand-held bluespace teleporter that can rip open portals to a random nearby location, or lock onto a teleporter with a selected teleportation beacon."
+	req_tech = list(TECH_BLUESPACE = 4, TECH_MATERIAL = 6)
+	materials = list(MATERIAL_GOLD = 2000, MATERIAL_DIAMOND = 1500, MATERIAL_URANIUM = 250)
+	build_path = /obj/item/hand_tele
```
Comment (NonQueueingMatt) [inline code/modules/research/designs/protolathe/tool_designs.dm:164]: I'm not a fan of R&D having access to these and printing an asston of them. I want these to be extremely expensive.

File: code/modules/research/designs/protolathe/weapon_designs.dm [hunk 4]
```diff
@@ -61,4 +61,40 @@
...
+	desc = "You can totally see in the dark now!"
+	req_tech = list(TECH_COMBAT = 6, TECH_ILLEGAL = 2)
+	materials = list(DEFAULT_WALL_MATERIAL = 1000, MATERIAL_GLASS = 3000, MATERIAL_GOLD = 1000)
+	build_path = /obj/item/clothing/glasses/night
+
+/datum/design/item/weapon/gravity_gun //Hello, Gordon!
+	desc = "This nifty gun disables the gravity in the area you shoot at. Use with caution."
+	req_tech = list(TECH_COMBAT = 4, TECH_BLUESPACE = 4)
+	materials = list(DEFAULT_WALL_MATERIAL = 5000, MATERIAL_GLASS = 500, MATERIAL_SILVER = 3000, MATERIAL_PHORON = 500)
+	build_path = /obj/item/gun/energy/gravity_gun
```
Comment (NonQueueingMatt) [inline code/modules/research/designs/protolathe/weapon_designs.dm:82]: Should probably be far more expensive.

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/modules/research/designs/protolathe/tool_designs.dm [hunk 5]
```diff
@@ -107,4 +107,82 @@ datum/design/item/tool/advanced_light_replacer
...
+	desc = "A high-luminosity flashlight for specialist duties."
+	req_tech = list(TECH_ENGINEERING = 3)
+	materials = list(DEFAULT_WALL_MATERIAL = 500, MATERIAL_GLASS = 500)
+	build_path = /obj/item/device/flashlight/heavy
+
+/datum/design/item/tool/tethering_device
+	desc = "A device used by explorers to keep track of partners by way of electro-tether."
+	req_tech = list(TECH_MAGNET = 5)
+	materials = list(DEFAULT_WALL_MATERIAL = 1500, MATERIAL_SILVER = 500)
+	build_path = /obj/item/tethering_device
```
Comment (Arrow768) [inline code/modules/research/designs/protolathe/tool_designs.dm:158]: These are all things that should be ordered via cargo.

File: code/modules/research/designs/protolathe/weapon_designs.dm [hunk 6]
```diff
@@ -61,4 +61,40 @@
...
+	desc = "A compact carbon-fiber respirator covering the mouth and nose to protect against the inhalation of smoke and other harmful gasses."
+	req_tech = list(TECH_BIO = 4, TECH_COMBAT = 4)
+	materials = list(MATERIAL_GLASS = 1000, DEFAULT_WALL_MATERIAL = 1000)
+	build_path = /obj/item/clothing/mask/gas/tactical
+
+/datum/design/item/weapon/nvg
+	desc = "You can totally see in the dark now!"
+	req_tech = list(TECH_COMBAT = 6, TECH_ILLEGAL = 2)
+	materials = list(DEFAULT_WALL_MATERIAL = 1000, MATERIAL_GLASS = 3000, MATERIAL_GOLD = 1000)
+	build_path = /obj/item/clothing/glasses/night
```
Comment (Arrow768) [inline code/modules/research/designs/protolathe/weapon_designs.dm:76]: Should also be ordered via cargo.

File: code/modules/research/designs/protolathe/weapon_designs.dm [hunk 7]
```diff
@@ -61,4 +61,40 @@
...
+	desc = "A device that can be attached to most firearms, providing a holographic display of the remaining ammunition to the user."
+	req_tech = list(TECH_BLUESPACE = 3, TECH_MATERIAL = 4, TECH_DATA = 4)
+	materials = list(DEFAULT_WALL_MATERIAL = 1000, MATERIAL_GLASS = 5000)
+	build_path = /obj/item/ammo_display
+
+/datum/design/item/weapon/maglight
+	desc = "A heavy flashlight designed for security personnel."
+	req_tech = list(TECH_COMBAT = 4)
+	materials = list(DEFAULT_WALL_MATERIAL = 200, MATERIAL_GLASS = 100)
+	build_path = /obj/item/device/flashlight/maglight
```
Comment (Arrow768) [inline code/modules/research/designs/protolathe/weapon_designs.dm:94]: That is also something for cargo.

File: code/modules/research/designs/protolathe/weapon_designs.dm [hunk 8]
```diff
@@ -61,4 +61,40 @@
...
+
+/datum/design/item/weapon/stunrod
+	desc = "A more-than-lethal weapon used to deal with high threat situations."
+	req_tech = list(TECH_COMBAT = 6, TECH_ILLEGAL = 2)
+	materials = list(DEFAULT_WALL_MATERIAL = 5000, MATERIAL_GLASS = 500, MATERIAL_SILVER = 1500)
+	build_path = /obj/item/melee/baton/stunrod
```
Comment (Arrow768) [inline code/modules/research/designs/protolathe/weapon_designs.dm:100]: That isnt something we want to have in regular use due to its 80 agony.

**Review by Arrow768 — DISMISSED** [review]
Autolathes can and should be used for common items with a very low technology level.
Cargo should be used for items that are reasonably sold by various corporations.
Research should take care of the uncommon high-tech items that arnt sold by corps.

The purpose of research is not to mass manufacture items for the station.

**Review by NonQueueingMatt — APPROVED** [review]

**Review by Alberyk — APPROVED** [review]

**Pull request merged**